src/main/java/org/candlepin/controller/PoolManager.java

```diff
@@ -268,13 +265,8 @@ void refreshPoolsWithRegeneration(SubscriptionServiceAdapter subAdapter,
 
             // Check if we've put any pools into a state in which they're referencing a product which no
             // longer belongs to the organization
-            List<String> pids = this.poolCurator.getPoolsUsingOrphanedProducts(resolvedOwner.getId());
-            if (pids != null && pids.size() > 0) {
-                log.error("One or more pools references a product which no longer belongs to its " +
-                    "organization: {}", pids);
 
-                throw new IllegalStateException("One or more pools was left in an undefined state: " + pids);
-            }
+            // TODO: Change this to ensure a pool isn't referencing a product from another org's namespace
```

Comment:
Will this be done as part of another task?

Reply:
This will be vetted and completed (if necessary) as part of CP-671

Reply:
Oops, misread the class file; thought it was in PoolResource.

Added the query and some tests for this logic.